x/collection/keeper/supply.go

```diff
@@ -317,19 +317,20 @@ func (k Keeper) ModifyContract(ctx sdk.Context, contractID string, operator sdk.
 		return err
 	}
 
-	modifiers := map[string]func(string){
-		collection.AttributeKeyName.String(): func(name string) {
+	modifiers := map[collection.AttributeKey]func(string){
+		collection.AttributeKeyName: func(name string) {
 			contract.Name = name
 		},
-		collection.AttributeKeyBaseImgURI.String(): func(uri string) {
+		collection.AttributeKeyBaseImgURI: func(uri string) {
 			contract.Name = uri
```

Comment:
```suggestion
			contract.BaseImgUri = uri
```
Shouldn't this be changed like this?

Reply:
Thanks. I didn't notice that. I should revisit entire modify logics on x/token and x/collection, and add the corresponding tests. It would be another bug fix PR.